.github/workflows/publish-docs-preview.yml

```diff
@@ -0,0 +1,58 @@
+name: Publish docs preview
+on:
+  pull_request_target:

```

Comment:
Using `pull_request_target` instead of `pull_request` so that I can access the secrets.
The code isn't checked out anyway.